src/utils/handleRequestWith.ts

```diff
@@ -70,6 +70,28 @@ export const handleRequestWith = (
       // Handle a scenario when there is no request handler
       // found for a given request.
       if (!handler) {
+        if (options.onUnhandledRequest === 'warn') {
+          // Produce a developer-friendly warning
+          console.warn(
+            `A request to ${req.url} was detected but not mocked because no request handler matching the URL exists.`,
+          )
+          return channel.send({ type: 'MOCK_NOT_FOUND' })
+        }
+
+        if (options.onUnhandledRequest === 'error') {
+          // Throw an exception
+
+          // throw new Error(`A request to ${req.url} was detected but not mocked because no request handler matching the URL exists.`)
+          return channel.send({ type: 'MOCK_NOT_FOUND' })
+        }
+
+        if (typeof options.onUnhandledRequest === 'function') {
+          options.onUnhandledRequest(req)
+          return channel.send({ type: 'MOCK_NOT_FOUND' })
+        }
+
+        // options.onUnhandledRequest === 'bypass'
```

Comment:
We can avoid `channel.send` duplication and `onUnhandledRequest` logic by creating the util described in a previous comment.

Reply:
Not sure of how much is the effort, but we should test this logic on both `worker.listen` and `server.start`

Reply:
Yes. Can you write the tests?

Reply:
Absolutely agree, there's no need to duplicate the `channel.send()` and any other possible logic in case of an unhandled request. You can think of the new feature as:

```ts
if (!response) {
  onUnhandledRequest(req, resolvedOptions) // analyzes "resolvedOptions.onUnhandledRequest"
  return channel.send({ type: 'MOCK_NOT_FOUND' })
}
```

Reply:
I can try!

Reply:
To clarify: I mean exactly what @cloud-walker suggested above: a custom function that encapsulates the logic (checks if the option value is a function, or one of the enums).

Reply:
> Yes. Would you be willing to write the tests?

How can contribute in this scenario? You give fork permissions to me? Or I should do another fork? 🤔

Reply:
I will give you permission to push to my fork

Reply:
I've merged the changes made by @kettanaito in the upstream repository so we're all up to date and I've given you access. Can you verify that it works? Haven't done this before.